MyList/MyList.cs

```
@@ -0,0 +1,222 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+
+namespace MyList
+{
+   public class MyList<T> : IEnumerator<T>, IEnumerable<T>
+   {
+      private const int initialCapacity = 10;
+
+      private T[] list = new T[initialCapacity];
+      private int index = -1;
+
+
+      public int Count { get; private set; }
+      public int Capacity
+      {
+         get
+         {
+            return list.Length;
+         }
+
+      }
+      public T Current
+      {
+         get
+         {
+            if (index < 0 || Count <= index)
+            {
+               return default(T);
```

In this case Current should throw an exception https://msdn.microsoft.com/en-us/library/system.collections.ienumerator.current(v=vs.110).aspx